src/jit/gentree.h

```diff
@@ -4240,9 +4240,17 @@ struct GenTreeAddrMode : public GenTreeOp
         return gtOp2;
     }
 
-    unsigned gtScale;  // The scale factor
-    unsigned gtOffset; // The offset to add
+    int Offset()
+    {
+        return static_cast<int>(gtOffset);
+    }
 
+    unsigned gtScale; // The scale factor
+#ifndef LEGACY_BACKEND
+private:
+#endif
+    unsigned gtOffset; // The offset to add
```

Comment:
Sorry I failed to respond to this. That strategy makes sense, though a comment to that effect would avoid confusion.